tests/test_schema_syntax.py

```diff
@@ -1346,22 +1419,70 @@ def test_eschema_syntax_link_12(self):
         };
         """
 
-    def test_eschema_syntax_property_13(self):
+    def test_eschema_syntax_link_13(self):
         """
         module test {
-            abstract link union extending except;
+            abstract link union {
+                extending except;
+            };
         };
         """
 
-    def test_eschema_syntax_property_14(self):
+    def test_eschema_syntax_link_14(self):
         """
         module test {
-            abstract link union extending intersect {
+            abstract link union {
+                extending intersect;
                 property except: str;
             };
         };
         """
 
+    def test_eschema_syntax_link_15(self):
+        """
+        module test {
+            abstract link foo {
```

Comment:
Hm, I think we could have skipped doing this for abstract links themselves and only done it for concrete ones. I don't have a super strong feeling though.